core/src/test/java/io/aiven/klaw/service/ServerConfigServiceTest.java

```diff
@@ -53,4 +94,498 @@ private void loginMock() {
+    assertThat(tenantConfig.getTenantModel().getRequestSchemaEnvironmentsList().get(0))
+        .isEqualTo("DEV_SCH");
+    assertThat(tenantConfig.getTenantModel().getRequestTopicsEnvironmentsList().get(0))
+        .isEqualTo("DEV");
+  }
+
+  @Test
+  @Order(13)
+  public void givenRequestForConfigWithNoneSet_returnCorrectConfig()
+      throws KlawException, JsonProcessingException {
+    stubValidateTests();
+    TenantConfig config = new TenantConfig();
+    config.setTenantModel(null);
+    Map<String, Map<String, String>> dbObject = new HashMap<>();
+    Map<String, String> map = new HashMap();
+
+    map.put("kwvalue", mapper.writeValueAsString(config));
+    map.put("kwkey", KLAW_TENANT_CONFIG);
+    map.put("kwdes", "Desc");
+    map.put("tenantid", "101");
+    dbObject.put(KLAW_TENANT_CONFIG, map);
+
+    when(managedb.getKwPropertiesMap(101)).thenReturn(dbObject);
+
+    // Execute
+    List<Map<String, String>> response = serverConfigService.getAllEditableProps();
+    TenantConfig tenantConfig =
+        mapper.readValue(response.get(0).get("kwvalue"), TenantConfig.class);
+    verify(managedb, times(1)).getKwPropertiesMap(101);
+    // assert that the order is as specified in getJsonString
+    assertThat(tenantConfig.getTenantModel()).isNull();
+  }
+
+  private Map<String, Map<String, String>> buildFullDbObject() throws JsonProcessingException {
+    // This object is created using IDs from stubValidateTests()
+    KwTenantConfigModel prop =
+        addKafkaTopicInformation(
+            new KwTenantConfigModel(), "1", Arrays.asList("1", "2", "3"), "1", "3", "2");
+    prop = addSchemaRegInformation(prop, Arrays.asList("7", "8", "9"), "7", "8", "9");
+    prop = addKafkaConnInformation(prop, Arrays.asList("5", "6", "4"), "6", "5", "4");
+
+    prop.setTenantName("default");
+    TenantConfig config = new TenantConfig();
+    config.setTenantModel(prop);
+    Map<String, Map<String, String>> dbObject = new HashMap<>();
+    Map<String, String> map = new HashMap();
+
+    map.put("kwvalue", mapper.writeValueAsString(config));
+    map.put("kwkey", KLAW_TENANT_CONFIG);
+    map.put("kwdes", "Desc");
+    map.put("tenantid", "101");
+    dbObject.put(KLAW_TENANT_CONFIG, map);
+
+    return dbObject;
+  }
+
+  private static KwPropertiesModel createKwPropertiesModel(String key, String value) {
+    KwPropertiesModel model = new KwPropertiesModel();
+    model.setKwKey(key);
+    model.setKwValue(value);
+    return model;
+  }
+
+  private void stubValidateTests() {
+    when(mailService.getUserName(any())).thenReturn("test");
+    when(commonUtilsService.getTenantId(any())).thenReturn(101);
+    when(managedb.getTenantMap())
+        .thenReturn(
+            new HashMap<Integer, String>() {
+              {
+                put(101, "default");
+              }
+            });
+
+    when(managedb.getKafkaEnvList(anyInt()))
+        .thenReturn(
+            List.of(
+                createEnv("DEV", "1", "kafka"),
```

Comment:
you might want to use KafkaClustersType enum here

Reply:
Updated.